### ISSUE TITLE
No rule found for atd.version.sexp

### ISSUE THREAD

**Issue by mjambon** [mjambon/atd#37]
The installation of the `atdgen` package fails when it's pinned to the local development folder (master branch).

```
$ opam pin add atdgen .
...
No rule found for atd.version.sexp
```
which comes from:
```
$ jbuilder build -p atdgen
No rule found for atd.version.sexp
```

The `atd` package uses an equivalent command (`jbuilder build -p atd`) but it succeeds. @rgrinberg, would you happen to know where `atd.version.sexp` comes from?

**Comment by mjambon** [comment]
FWIW `jbuilder build -p atdgen` leaves a file `_build/default/atdgen.version.sexp` that contains `(v1.12.0)`.

**Comment by rgrinberg** [comment]
It's probably b/c we are using the atd version for the various version strings. The rule to generate that version string only belongs to atd itself.

@diml shouldn't jbuilder be able to generate the version sexp for external packages as well? (If i'm interpreting this issue right).

@mjambon btw, I think the best fix is to use topkg and rely on `$ jbuilder subst` to enter correct version numbers everywhere. Are you fine with making releases using topkg? You can still make manual submissions to opam btw, but the build instructions for opam will have the extra `$ jbuilder subst` step. Another plus of this setup is that it will include the git commit for pinned packages in the version number.

**Comment by rgrinberg** [comment]
PS: the consequence of using `$ jbuilder subst` would mean that the version will always the version of the package itself and not the version of atd. But I think that's probably for the better anyway. Ideally, we'd have something like `Atd.version` `Atdgen_runtime.version` etc. all returning their respective version numbers.

**Comment by mjambon** [comment]
I haven't used topkg but it looks reasonable. Go for it.

**Comment by mjambon** [comment]
Fixed with #38, which uses topkg for versioning.
